### PR TITLE
fix: Fix Code style issues to match Java standards

### PR DESCRIPTION
**Pull request by ashwithpoojary98**
## Change list

- Added HTML tags to doc links
- Added a private constructor to restrict the creation of new objects of the class
- Removed extra parentheses from lambda expressions
- Renamed class and method variables to be more descriptive
- Added override annotations to methods that override the parent class
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [x] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

To meet Java code standards, I made a few changes to method and class-level variables:
Added a Lombok no-arg constructor with private access to restrict the user from creating a new object of the class, since this class consists of all static methods.
Added override annotations to improve the readability of methods and removed additional code from lambda expressions.

### CODE REVIEW

File: src/main/java/io/appium/java_client/imagecomparison/ComparisonResult.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public abstract class ComparisonResult {
 
     @Getter(AccessLevel.PROTECTED) private final Map<String, Object> commandResult;
 
-    public ComparisonResult(Map<String, Object> commandResult) {
+    protected ComparisonResult(Map<String, Object> commandResult) {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/imagecomparison/ComparisonResult.java:37]: this is a breaking change

File: src/main/java/io/appium/java_client/pagefactory/interceptors/InterceptorOfAListOfElements.java [hunk 2]
```diff
@@ -29,7 +29,7 @@
 public abstract class InterceptorOfAListOfElements implements MethodCallListener {
     protected final ElementLocator locator;
 
-    public InterceptorOfAListOfElements(@Nullable ElementLocator locator) {
+    protected InterceptorOfAListOfElements(@Nullable ElementLocator locator) {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/pagefactory/interceptors/InterceptorOfAListOfElements.java:32]: this is a breaking change
same below

File: src/main/java/io/appium/java_client/remote/options/SystemScript.java [hunk 3]
```diff
@@ -20,10 +20,10 @@
 import java.util.Optional;
 
 public abstract class SystemScript<T extends SystemScript<T>> extends BaseMapOptionData<T> {
-    public SystemScript() {
+    protected SystemScript() {
     }
 
-    public SystemScript(Map<String, Object> options) {
+    protected SystemScript(Map<String, Object> options) {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/remote/options/SystemScript.java:26]: same here

**Comment by mykola-mokhnach** [comment]
Thank you @ashwithpoojary98 
Please rename the PR to follow https://www.conventionalcommits.org/en/v1.0.0/ 
Also please make sure you have signed the CLA

File: src/main/java/io/appium/java_client/imagecomparison/ComparisonResult.java (hunk 1 above)
Comment (ashwithpoojary98) [reply src/main/java/io/appium/java_client/imagecomparison/ComparisonResult.java]: reverted the changes to public

File: src/main/java/io/appium/java_client/pagefactory/interceptors/InterceptorOfAListOfElements.java (hunk 2 above)
Comment (ashwithpoojary98) [reply src/main/java/io/appium/java_client/pagefactory/interceptors/InterceptorOfAListOfElements.java]: reverted to public

File: src/main/java/io/appium/java_client/remote/options/SystemScript.java (hunk 3 above)
Comment (ashwithpoojary98) [reply src/main/java/io/appium/java_client/remote/options/SystemScript.java]: reverted to public

**Comment by ashwithpoojary98** [comment]
@mykola-mokhnach updated the code as per the comment Please review

File: src/main/java/io/appium/java_client/remote/options/SystemScript.java [hunk 4]
```diff
@@ -20,7 +20,7 @@
 import java.util.Optional;
 
 public abstract class SystemScript<T extends SystemScript<T>> extends BaseMapOptionData<T> {
-    public SystemScript() {
+    protected SystemScript() {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/remote/options/SystemScript.java:23]: this is a breaking change

Comment (ashwithpoojary98) [reply src/main/java/io/appium/java_client/remote/options/SystemScript.java]: reverted the changes

**Comment by ashwithpoojary98** [comment]
updated the workflow issue

**Comment by ashwithpoojary98** [comment]
Updated workflow issue

File: src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java [hunk 5]
```diff
@@ -39,10 +39,10 @@
 import static io.appium.java_client.remote.MobilePlatform.WINDOWS;
 
 /**
- * It is the basic handler of Appium-specific page object annotations
+ * It is the basic handler of Appium-specific page object annotations.
  * About the Page Object design pattern please read these documents:
- * - https://code.google.com/p/selenium/wiki/PageObjects
- * - https://code.google.com/p/selenium/wiki/PageFactory
+ * - <a href="https://code.google.com/p/selenium/wiki/PageObjects">https://code.google.com/p/selenium/wiki/PageObjects</a>
+ * - <a href="https://code.google.com/p/selenium/wiki/PageFactory">https://code.google.com/p/selenium/wiki/PageFactory</a>
```
Comment (valfirst) [inline src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java:45]:
```suggestion
 * - <a href="https://github.com/SeleniumHQ/selenium/wiki/PageFactory">Selenium Page Factory</a>
```

File: src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java [hunk 6]
```diff
@@ -39,10 +39,10 @@
 import static io.appium.java_client.remote.MobilePlatform.WINDOWS;
 
 /**
- * It is the basic handler of Appium-specific page object annotations
+ * It is the basic handler of Appium-specific page object annotations.
  * About the Page Object design pattern please read these documents:
- * - https://code.google.com/p/selenium/wiki/PageObjects
- * - https://code.google.com/p/selenium/wiki/PageFactory
+ * - <a href="https://code.google.com/p/selenium/wiki/PageObjects">https://code.google.com/p/selenium/wiki/PageObjects</a>
```
Comment (valfirst) [inline src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java:44]:
```suggestion
 * - <a href="https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/">Selenium Page Object models</a>
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by SrinivasanTarget — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Comment by ashwithpoojary98** [comment]
All checks have passed!!

**Pull request merged**